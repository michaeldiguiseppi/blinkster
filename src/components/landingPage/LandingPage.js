import React, { Component } from 'react';
import Car from '../car/Car';

require('./css/LandingPage.css');

export class LandingPage extends Component {
    constructor(props) {
        super(props);
        this._fetchCars = this._fetchCars.bind(this);
        this._renderLoading = this._renderLoading.bind(this);
        this._setupCarsData = this._setupCarsData.bind(this);
    }

    componentDidMount() {
        this._fetchCars();
    }

    _fetchCars() {
        let { carsActions } = this.props;
        return carsActions.fetchCars();
    }

    _setupCarsData() {
        if (this.props.cars && this.props.cars.cars && this.props.cars.cars.length > 0) {
            return this.props.cars.cars.map((car, index) => {
                return (
                    <Car 
                        car={ car } 
                        key={ index }
                        index={ index }
                    />
                )
            });
        }
    }

    _renderLoading() {
        return (
            <div className="jumbotron">
                <h1 className="text-center">Please wait, Loading...</h1>
            </div>
        )
    }

    render() {
        let { cars } = this.props;
        return (
            <div>
            { cars.isLoading ? this._renderLoading() : 
                <div className="table-responsive">
                    <table className="table table-hover overflow">
                        <thead className="text-center">
                            <tr>
                                <th scope="col">Year</th>
                                <th scope="col">Make</th>
                                <th scope="col">Model</th>
                                <th scope="col">Mileage</th>
                                <th scope="col">Drivetrain</th>
                            </tr>
                        </thead>
                        <tbody className="text-center">
                            { this._setupCarsData() }
                        </tbody>
                    </table>
                </div>
                 }
            </div>
        )
    }
}

export default LandingPage;