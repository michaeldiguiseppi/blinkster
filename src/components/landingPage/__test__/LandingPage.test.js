import React from 'react';
import LandingPage from '../LandingPage';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });


describe('<LandingPage /> component', () => {

    const defaultProps = {
        cars: {
            isLoading: false,
            cars: []
        },
        carsActions: {
            fetchCars: () => {}
        }
    }

    it('should render without breaking', () => {
        const component = shallow(<LandingPage {...defaultProps} />);
        expect(component.find("table").exists()).toEqual(true);
    });

    it('should match the snapshot', () => {
        const tree = renderer
            .create(<LandingPage {...defaultProps} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});