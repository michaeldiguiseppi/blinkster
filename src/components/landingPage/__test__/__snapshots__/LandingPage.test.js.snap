// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingPage /> component should match the snapshot 1`] = `
<div>
  <div
    className="search-input form-control"
  >
    <input
      className={undefined}
      onChange={[Function]}
      placeholder="Search by year, make, model, or drivetrain..."
      type="search"
      value=""
    />
  </div>
  <div
    className="table-responsive"
  >
    <table
      className="table table-hover overflow"
    >
      <thead
        className="text-center"
      >
        <tr>
          <th
            onClick={[Function]}
            scope="col"
          >
            year
             
          </th>
          <th
            onClick={[Function]}
            scope="col"
          >
            make
             
          </th>
          <th
            onClick={[Function]}
            scope="col"
          >
            model
             
          </th>
          <th
            onClick={[Function]}
            scope="col"
          >
            mileage
             
          </th>
          <th
            onClick={[Function]}
            scope="col"
          >
            drivetrain
             
          </th>
        </tr>
      </thead>
      <tbody
        className="text-center"
      >
        <tr>
          <th
            scope="row"
          >
            2013
          </th>
          <td>
            Kia
          </td>
          <td>
            Optima
          </td>
          <td>
            24235
          </td>
          <td
            className="veh-drivetrain"
          >
            FWD
          </td>
          <td>
            <a
              href="/cars/2013/Kia/Optima"
            >
              <button
                className="btn btn-outline-info"
                type="button"
              >
                Details
              </button>
            </a>
          </td>
        </tr>
        <tr>
          <th
            scope="row"
          >
            2013
          </th>
          <td>
            Hyundai
          </td>
          <td>
            Accent
          </td>
          <td>
            21587
          </td>
          <td
            className="veh-drivetrain"
          >
            FWD
          </td>
          <td>
            <a
              href="/cars/2013/Hyundai/Accent"
            >
              <button
                className="btn btn-outline-info"
                type="button"
              >
                Details
              </button>
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
